### ISSUE TITLE
raspberry pi 4: getting errors

### ISSUE THREAD

**Issue by pakair** [mikaku/Monitorix#371]
I installed monitorix 3.12.0 (21-Feb-2020) on a raspberry pi 4 (8GB), in the config file, line 116 I put raspberrypi = y and restarted the service. I get two errors,

ERROR: while graphing /var/lib/monitorix/www//imgs/raspberrypi1.1day.png: opening '/var/lib/monitorix/raspberrypi.rrd': No such file or directory ERROR: while graphing /var/lib/monitorix/www//imgs/raspberrypi1z.1day.png: opening '/var/lib/monitorix/raspberrypi.rrd': No such file or directory 

ERROR:  while graphing /var/lib/monitorix/www//imgs/raspberrypi2.1day.png:  opening '/var/lib/monitorix/raspberrypi.rrd': No such file or directory ERROR: while graphing  /var/lib/monitorix/www//imgs/raspberrypi2z.1day.png: opening  '/var/lib/monitorix/raspberrypi.rrd': No such file or directory        ERROR: while graphing  /var/lib/monitorix/www//imgs/raspberrypi3.1day.png: opening  '/var/lib/monitorix/raspberrypi.rrd': No such file or directory ERROR: while graphing  /var/lib/monitorix/www//imgs/raspberrypi3z.1day.png: opening  '/var/lib/monitorix/raspberrypi.rrd': No such file or directory

If I change raspberrypi = n all graphs look good. Maybe I should not use that option.

**Comment by mikaku** [comment]
Please, paste here the Monitorix log file right after you started Monitorix.
Thanks.

**Comment by pakair** [comment]
Fri Dec 10 04:21:36 2021 - Starting Monitorix version 3.12.0 (pid 28348).
Fri Dec 10 04:21:36 2021 - Loaded main configuration file '/etc/monitorix/monitorix.conf'.
Fri Dec 10 04:21:36 2021 - Loading extra configuration file '/etc/monitorix/conf.d/00-debian.conf'.
Can't exec "iptables": No such file or directory at /usr/bin/monitorix line 672.
Use of uninitialized value $ipv in scalar chomp at /usr/bin/monitorix line 673.
Use of uninitialized value $ipv in substitution (s///) at /usr/bin/monitorix line 674.
Use of uninitialized value $ipv in split at /usr/bin/monitorix line 675.
Missing argument in sprintf at /usr/bin/monitorix line 675.
Missing argument in sprintf at /usr/bin/monitorix line 675.
Missing argument in sprintf at /usr/bin/monitorix line 675.
Can't exec "iptables": No such file or directory at /usr/lib/monitorix/Monitorix.pm line 333.
Can't exec "iptables": No such file or directory at /usr/lib/monitorix/Monitorix.pm line 351.
Can't exec "ip6tables": No such file or directory at /usr/lib/monitorix/Monitorix.pm line 375.
Can't exec "ip6tables": No such file or directory at /usr/lib/monitorix/Monitorix.pm line 393.
Can't exec "iptables": No such file or directory at /usr/lib/monitorix/Monitorix.pm line 415.
Can't exec "ip6tables": No such file or directory at /usr/lib/monitorix/Monitorix.pm line 437.
Fri Dec 10 04:21:36 2021 - Initializing graphs.
Fri Dec 10 04:21:36 2021 - fs::fs_init: Unable to detect the device name of '/', I/O stats won't be shown in graph. If this is really a mount point then consider using <devmap> to map it manually to a device name.
Fri Dec 10 04:21:36 2021 - fs::fs_init: Unable to detect the device name of 'swap', I/O stats won't be shown in graph. If this is really a mount point then consider using <devmap> to map it manually to a device name.
Can't exec "iptables": No such file or directory at /usr/lib/monitorix/port.pm line 169.
Can't exec "iptables": No such file or directory at /usr/lib/monitorix/port.pm line 170.
Can't exec "iptables": No such file or directory at /usr/lib/monitorix/port.pm line 169.
Can't exec "iptables": No such file or directory at /usr/lib/monitorix/port.pm line 170.
Can't exec "iptables": No such file or directory at /usr/lib/monitorix/port.pm line 169.
Can't exec "iptables": No such file or directory at /usr/lib/monitorix/port.pm line 170.
Can't exec "iptables": No such file or directory at /usr/lib/monitorix/port.pm line 169.
Can't exec "iptables": No such file or directory at /usr/lib/monitorix/port.pm line 170.
Can't exec "iptables": No such file or directory at /usr/lib/monitorix/port.pm line 169.
Can't exec "iptables": No such file or directory at /usr/lib/monitorix/port.pm line 170.
Can't exec "iptables": No such file or directory at /usr/lib/monitorix/port.pm line 169.
Can't exec "iptables": No such file or directory at /usr/lib/monitorix/port.pm line 170.
Can't exec "iptables": No such file or directory at /usr/lib/monitorix/port.pm line 169.
Can't exec "iptables": No such file or directory at /usr/lib/monitorix/port.pm line 170.
Can't exec "iptables": No such file or directory at /usr/lib/monitorix/port.pm line 169.
Can't exec "iptables": No such file or directory at /usr/lib/monitorix/port.pm line 170.
Can't exec "iptables": No such file or directory at /usr/lib/monitorix/port.pm line 169.
Can't exec "iptables": No such file or directory at /usr/lib/monitorix/port.pm line 170.
Fri Dec 10 04:21:36 2021 - Generating the 'index.html' file.
Fri Dec 10 04:21:36 2021 - Started HTTP built-in server (pid 28391).
Fri Dec 10 04:21:36 2021 - Ok, ready.
Use of uninitialized value in concatenation (.) or string at /usr/lib/monitorix/port.pm line 322.
Use of uninitialized value in concatenation (.) or string at /usr/lib/monitorix/port.pm line 322.
Use of uninitialized value in concatenation (.) or string at /usr/lib/monitorix/port.pm line 322.
Use of uninitialized value in concatenation (.) or string at /usr/lib/monitorix/port.pm line 322.
Use of uninitialized value in concatenation (.) or string at /usr/lib/monitorix/port.pm line 322.
Use of uninitialized value in concatenation (.) or string at /usr/lib/monitorix/port.pm line 322.
Use of uninitialized value in concatenation (.) or string at /usr/lib/monitorix/port.pm line 322.
Use of uninitialized value in concatenation (.) or string at /usr/lib/monitorix/port.pm line 322.
Use of uninitialized value in concatenation (.) or string at /usr/lib/monitorix/port.pm line 322.
Use of uninitialized value in concatenation (.) or string at /usr/lib/monitorix/port.pm line 322.
Use of uninitialized value in concatenation (.) or string at /usr/lib/monitorix/port.pm line 322.
Use of uninitialized value in concatenation (.) or string at /usr/lib/monitorix/port.pm line 322.
Use of uninitialized value in concatenation (.) or string at /usr/lib/monitorix/port.pm line 322.
Use of uninitialized value in concatenation (.) or string at /usr/lib/monitorix/port.pm line 322.
Use of uninitialized value in concatenation (.) or string at /usr/lib/monitorix/port.pm line 322.
Use of uninitialized value in concatenation (.) or string at /usr/lib/monitorix/port.pm line 322.
Use of uninitialized value in concatenation (.) or string at /usr/lib/monitorix/port.pm line 322.
Use of uninitialized value in concatenation (.) or string at /usr/lib/monitorix/port.pm line 322.
Use of uninitialized value in concatenation (.) or string at /usr/lib/monitorix/port.pm line 322.
Use of uninitialized value in concatenation (.) or string at /usr/lib/monitorix/port.pm line 322.
Use of uninitialized value in concatenation (.) or string at /usr/lib/monitorix/port.pm line 322.
Use of uninitialized value in concatenation (.) or string at /usr/lib/monitorix/port.pm line 322.
Use of uninitialized value in concatenation (.) or string at /usr/lib/monitorix/port.pm line 322.
Use of uninitialized value in concatenation (.) or string at /usr/lib/monitorix/port.pm line 322.
Use of uninitialized value in concatenation (.) or string at /usr/lib/monitorix/port.pm line 322.
Use of uninitialized value in concatenation (.) or string at /usr/lib/monitorix/port.pm line 322.
Use of uninitialized value in concatenation (.) or string at /usr/lib/monitorix/port.pm line 322.
Use of uninitialized value in concatenation (.) or string at /usr/lib/monitorix/port.pm line 322.
Use of uninitialized value in concatenation (.) or string at /usr/lib/monitorix/port.pm line 322.
Use of uninitialized value in concatenation (.) or string at /usr/lib/monitorix/port.pm line 322.
Use of uninitialized value in concatenation (.) or string at /usr/lib/monitorix/port.pm line 322.
Use of uninitialized value in concatenation (.) or string at /usr/lib/monitorix/port.pm line 322.
Use of uninitialized value in concatenation (.) or string at /usr/lib/monitorix/port.pm line 322.
Use of uninitialized value in concatenation (.) or string at /usr/lib/monitorix/port.pm line 322.
Use of uninitialized value in concatenation (.) or string at /usr/lib/monitorix/port.pm line 322.
Use of uninitialized value in concatenation (.) or string at /usr/lib/monitorix/port.pm line 322.
Fri Dec 10 04:22:00 2021 - ERROR: while updating /var/lib/monitorix/port.rrd: /var/lib/monitorix/port.rrd: Function update_pdp_prep, case DST_GAUGE - Cannot convert '' to float
Fri Dec 10 04:21:37 2021 - WARNING: the HTTP built-in server has authentication disabled.
HTTPServer: You can connect to your server at http://localhost:8080/
$

**Comment by mikaku** [comment]
> ` Fri Dec 10 04:21:36 2021 - Loading extra configuration file '/etc/monitorix/conf.d/00-debian.conf'.`

Are you aware that Monitorix is loading an extra configuration file that can override your settings?


> `Can't exec "iptables": No such file or directory at /usr/lib/monitorix/port.pm line 169.`

If you want to get rid of these messages, just install `iptables`.

I don't see anything related to Raspberry Pi, perhaps if you can provide your configuration file ...

**Comment by pakair** [comment]
I installed iptables to I dont get that message. Let me try again, I change raspberrypi = y in the config file, restart service

Fri Dec 10 04:55:33 2021 - Starting Monitorix version 3.12.0 (pid 30946).
Fri Dec 10 04:55:33 2021 - Loaded main configuration file '/etc/monitorix/monitorix.conf'.
Fri Dec 10 04:55:33 2021 - Loading extra configuration file '/etc/monitorix/conf.d/00-debian.conf'.
Fri Dec 10 04:55:33 2021 - Initializing graphs.
Fri Dec 10 04:55:33 2021 - fs::fs_init: Unable to detect the device name of '/', I/O stats won't be shown in graph. If this is really a mount point then consider using <devmap> to map it manually to a device name.
Fri Dec 10 04:55:33 2021 - fs::fs_init: Unable to detect the device name of 'swap', I/O stats won't be shown in graph. If this is really a mount point then consider using <devmap> to map it manually to a device name.
Fri Dec 10 04:55:33 2021 - raspberrypi::raspberrypi_init: '/opt/vc/bin/vcgencmd' not found or is not executable. No such file or directory
Fri Dec 10 04:55:33 2021 - Generating the 'index.html' file.

pi@zero:~ $ whereis vcgencmd
vcgencmd: /usr/bin/vcgencmd /usr/share/man/man1/vcgencmd.1.gz

here's a screenshot of the errors, https://imgur.com/a/2bHhDJc

**Comment by mikaku** [comment]
> ` Fri Dec 10 04:55:33 2021 - raspberrypi::raspberrypi_init: '/opt/vc/bin/vcgencmd' not found or is not executable. No such file or directory`

Looks like your Monitorix is searching the `vcgencmd` command in a different location.
Please, just change the path of that command in your configuration file:

```
<raspberrypi>
        cmd = /usr/bin/vcgencmd
[...]
```

and restart Monitorix.

**Comment by pakair** [comment]
Now, I do not see the errors, I see a graph ... thank you.
![image](https://user-images.githubusercontent.com/17964012/145572318-5381f0bf-0483-4eb0-aa9e-097930cd764b.png)

**Comment by mikaku** [comment]
Perfect!
You should start seeing statistics in a few minutes.

I'll change the default path `/opt/vc/bin/vcgencmd` by `/usr/bin/vcgencmd` in the next Monitorix version.
Thanks for your feedback.

**Comment by pakair** [comment]
No worries, I'll close the ticket.

**Comment by mikaku** [comment]
> I'll change the default path /opt/vc/bin/vcgencmd by /usr/bin/vcgencmd in the next Monitorix version.

Looks like other distributions are still using the default path, so I'll keep this path by now.
